rust/cubesql/cubesql/src/compile/rewrite/converter.rs

```diff
@@ -2441,3 +2441,29 @@ fn replace_qualified_col_with_flat_name_if_missing(
         })
         .collect::<Result<Vec<_>, _>>()
 }
+
+/// Recursively normalize all Column expressions in a list of expression trees
+fn normalize_cols(
+    exprs: impl IntoIterator<Item = impl Into<Expr>>,
+    plan: &LogicalPlan,
+) -> Result<Vec<Expr>, CubeError> {
+    exprs
+        .into_iter()
+        .map(|e| normalize_col(e.into(), plan))
+        .collect()
+}
+
+/// Recursively call [`df_normalize_col`] on all Column expressions
+/// in the `expr` expression tree, realiasing the expressions if the name is different.
+fn normalize_col(expr: Expr, plan: &LogicalPlan) -> Result<Expr, CubeError> {
+    if let Expr::Alias(_, _) = expr {
+        return Ok(df_normalize_col(expr, plan)?);
+    }
+    let original_expr_name = expr_name(&expr)?;
+    let mut normalized_expr = df_normalize_col(expr, plan)?;
+    let normalized_expr_name = expr_name(&normalized_expr)?;
+    if original_expr_name != normalized_expr_name {
```

Comment:
Just for reference - DataFusion provides some tools in public API for this problem:

`rewrite_preserving_name`, `name_for_alias` and `alias_if_changed` starting from 27.0.0

https://docs.rs/datafusion/27.0.0/datafusion/logical_expr/expr_rewriter/fn.rewrite_preserving_name.html

https://docs.rs/datafusion/27.0.0/datafusion/logical_expr/enum.Expr.html#method.name_for_alias

https://docs.rs/datafusion/27.0.0/datafusion/logical_expr/enum.Expr.html#method.alias_if_changed

https://github.com/apache/datafusion/commit/feb4b8fbe19311aa8f1153662f3f75d5235e8fcb

`NamePreserver`  starting from 38.0.0

https://docs.rs/datafusion/38.0.0/datafusion/optimizer/utils/struct.NamePreserver.html

https://github.com/apache/datafusion/commit/e161cd65bc910a166ead9d93a17295c25cc08a3c